.github/workflows/check-backend-on-pr.yml

```diff
@@ -0,0 +1,114 @@
+name: "Check Backend on Pull Request"
+
+on:
+  pull_request:
+    branches:
+      - develop
+      - main
+    paths:
+      - "backend/**/*.rs"
+      - "backend/**/Cargo.toml"
+      - "backend/**/Cargo.lock"
+      - "backend/rust-toolchain.toml"
+      - "backend/rustfmt.toml"
+      - ".github/workflows/*.yml"
+  push:
+    branches:
+      - develop
+
+# Ensures only one build is run per branch, unless pushing to develop
```

Comment:
I think we can only build the latest on develop too. What benefit do you see in not cancelling the previous run?

Reply:
Not much really, It was more of an inheritance of the ajuna-pallets worklfow